parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -622,10 +622,12 @@ configGPUDrivers() {
   echo blacklist nouveau >>/etc/modprobe.d/blacklist.conf
   retrycmd_no_stats 120 5 25 update-initramfs -u || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
   wait_for_apt_locks
-  dpkg -i $(ls ${APT_CACHE_DIR}libnvidia-container1*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
-  dpkg -i $(ls ${APT_CACHE_DIR}libnvidia-container-tools*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
-  dpkg -i $(ls ${APT_CACHE_DIR}nvidia-container-toolkit*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
-  dpkg -i $(ls ${APT_CACHE_DIR}nvidia-container-runtime*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
+  for apt_package in $NVIDIA_PACKAGES; do
+    dpkg -i $(ls ${PERMANENT_CACHE_DIR}${apt_package}*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
+    dpkg -i $(ls ${PERMANENT_CACHE_DIR}${apt_package}*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
+    dpkg -i $(ls ${PERMANENT_CACHE_DIR}${apt_package}*) || exit {{GetCSEErrorCode "ERR_GPU_DRIVERS_CONFIG"}}
```

Comment:
Why the 3 dpkg -i of the same operation but failing after any one fails?

Why use `dpkg -i $(ls ${PERMANENT_CACHE_DIR}${apt_package}*)` when `dpkg -i ${PERMANENT_CACHE_DIR}${apt_package}*` would work equally the same and not cause problems with files that may have spaces in the name (not that there should be any of those, but...)

Reply:
Is this due to some retries to address ordering?  If so, why would that work if you exit after any error?

Reply:
It's not the same operation, `$NVIDIA_PACKAGES` is a "space-delimited string enumerator" with the value:

`"libnvidia-container1 libnvidia-container-tools nvidia-container-toolkit"`

All of those libs are required, so if any one of them fails to install we want to fail.

Reply:
You have a loop to do all of them and then you do each one 3 times?
Note how lines 626, 627, and 628 are all the same - so you are doing the same thing 3 times, not 3 different packages.